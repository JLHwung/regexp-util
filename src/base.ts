export abstract class Base {
  // tslint:disable-next-line:naming-convention
  public abstract toString(): string;
  // tslint:disable-next-line:naming-convention
  public toRegExp(flags?: string) {
    return new RegExp(this.toString(), flags);
  }
}
