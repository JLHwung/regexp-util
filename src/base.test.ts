import { Base } from './base';

class Test extends Base {
  public data = '123';
  // tslint:disable-next-line:naming-convention
  public toString() {
    return this.data;
  }
}

test('toRegExp: default', () => {
  expect(new Test().toRegExp()).toMatchSnapshot();
});

test('toRegExp: accept flags', () => {
  expect(new Test().toRegExp('u')).toMatchSnapshot();
});
