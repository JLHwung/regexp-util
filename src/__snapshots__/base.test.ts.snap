// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toRegExp: accept flags 1`] = `/123/u`;

exports[`toRegExp: default 1`] = `/123/`;
